covalent/_workflow/transportable_object.py

```diff
@@ -0,0 +1,420 @@
+        object_dict = json.loads(json_string)
+        return TransportableObject.from_dict(object_dict)
+
+    @staticmethod
+    def make_transportable(obj: Any) -> "TransportableObject":
+        """Make an object transportable.
+
+        Args:
+            obj: The object to make transportable.
+
+        Returns:
+            Transportable object.
+
+        """
+        if isinstance(obj, TransportableObject):
+            return obj
+        else:
+            return TransportableObject(obj)
+
+    @staticmethod
+    def deserialize(
+        serialized: bytes, *, header_only: bool = False, string_only: bool = False
+    ) -> "TransportableObject":
+        """Deserialize the transportable object.
+
+        Args:
+            data: Serialized transportable object
+
+        Returns:
+            object: The deserialized transportable object.
+
+        """
+        ar = _TOArchive.load(serialized, header_only, string_only)
+        return _from_archive(ar)
+
+    @staticmethod
+    def deserialize_list(collection: list) -> list:
+        """
+        Recursively deserializes a list of TransportableObjects. More
+        precisely, `collection` is a list, each of whose entries is
+        assumed to be either a `TransportableObject`, a list, or dict`
+
+        Args:
+            collection: A list of TransportableObjects.
+
+        Returns:
+            A list of deserialized objects.
+
+        """
+        new_list = []
+        for item in collection:
+            if isinstance(item, TransportableObject):
+                new_list.append(item.get_deserialized())
+            elif isinstance(item, list):
+                new_list.append(TransportableObject.deserialize_list(item))
+            elif isinstance(item, dict):
+                new_list.append(TransportableObject.deserialize_dict(item))
+            else:
+                raise TypeError("Couldn't deserialize collection")
+        return new_list
+
+    @staticmethod
+    def deserialize_dict(collection: dict) -> dict:
+        """
+        Recursively deserializes a dict of TransportableObjects. More
+        precisely, `collection` is a dict, each of whose entries is
+        assumed to be either a `TransportableObject`, a list, or dict`
+
+        Args:
+            collection: A dictionary of TransportableObjects.
+
+        Returns:
+            A dictionary of deserialized objects.
+
+        """
+        new_dict = {}
+        for k, item in collection.items():
+            if isinstance(item, TransportableObject):
+                new_dict[k] = item.get_deserialized()
+            elif isinstance(item, list):
+                new_dict[k] = TransportableObject.deserialize_list(item)
+            elif isinstance(item, dict):
+                new_dict[k] = TransportableObject.deserialize_dict(item)
+            else:
+                raise TypeError("Couldn't deserialize collection")
+        return new_dict
+
+
+def _to_archive(to: TransportableObject) -> _TOArchive:
```

Comment:
Would make sense for this to be a static method (non private) of `TransportableObject`

Reply:
This is meant to be used via the `serialize/deserialize` methods in the TransportableObject. It's not really intended to be used directly.

Reply:
Moved this to Archived Transportable Object